packages/@aws-cdk/aws-ecs-patterns/lib/fargate/queue-processing-fargate-service.ts

```diff
@@ -121,20 +122,36 @@ export class QueueProcessingFargateService extends QueueProcessingServiceBase {
 
     // Create a Fargate service with the previously defined Task Definition and configure
     // autoscaling based on cpu utilization and number of messages visible in the SQS queue.
-    this.service = new FargateService(this, 'QueueProcessingFargateService', {
-      cluster: this.cluster,
-      desiredCount: this.desiredCount,
-      taskDefinition: this.taskDefinition,
-      serviceName: props.serviceName,
-      minHealthyPercent: props.minHealthyPercent,
-      maxHealthyPercent: props.maxHealthyPercent,
-      propagateTags: props.propagateTags,
-      enableECSManagedTags: props.enableECSManagedTags,
-      platformVersion: props.platformVersion,
-      deploymentController: props.deploymentController,
-      securityGroups: props.securityGroups,
-      vpcSubnets: props.taskSubnets,
-    });
+    if (!this.node.tryGetContext(cxapi.ECS_REMOVE_DEFAULT_DESIRED_COUNT)) {
+      this.service = new FargateService(this, 'QueueProcessingFargateService', {
+        cluster: this.cluster,
+        desiredCount: this.desiredCount,
+        taskDefinition: this.taskDefinition,
+        serviceName: props.serviceName,
+        minHealthyPercent: props.minHealthyPercent,
+        maxHealthyPercent: props.maxHealthyPercent,
+        propagateTags: props.propagateTags,
+        enableECSManagedTags: props.enableECSManagedTags,
+        platformVersion: props.platformVersion,
+        deploymentController: props.deploymentController,
+        securityGroups: props.securityGroups,
+        vpcSubnets: props.taskSubnets,
+      });
+    } else {
+      this.service = new FargateService(this, 'QueueProcessingFargateService', {
+        cluster: this.cluster,
+        taskDefinition: this.taskDefinition,
+        serviceName: props.serviceName,
+        minHealthyPercent: props.minHealthyPercent,
+        maxHealthyPercent: props.maxHealthyPercent,
+        propagateTags: props.propagateTags,
+        enableECSManagedTags: props.enableECSManagedTags,
+        platformVersion: props.platformVersion,
+        deploymentController: props.deploymentController,
+        securityGroups: props.securityGroups,
+        vpcSubnets: props.taskSubnets,
+      });
+    }
```

Comment:
```suggestion
        const desiredCount = this.node.tryGetContext(cxapi.ECS_REMOVE_DEFAULT_DESIRED_COUNT) ? undefined : this.desiredCount;
    this.service = new FargateService(this, 'QueueProcessingFargateService', {
      cluster: this.cluster,
      desiredCount: desiredCount,
      taskDefinition: this.taskDefinition,
      serviceName: props.serviceName,
      minHealthyPercent: props.minHealthyPercent,
      maxHealthyPercent: props.maxHealthyPercent,
      propagateTags: props.propagateTags,
      enableECSManagedTags: props.enableECSManagedTags,
      platformVersion: props.platformVersion,
      deploymentController: props.deploymentController,
      securityGroups: props.securityGroups,
      vpcSubnets: props.taskSubnets,
      assignPublicIp: props.assignPublicIp,
    });

```

Reply:
I was trying to separate out the logic so that when the time comes for the next release, it's just easier to clean up since all you need to do is delete the if block. Let me know if you disagree